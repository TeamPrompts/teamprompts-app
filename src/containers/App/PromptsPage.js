import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import Filters from '../../components/Filters/Filters';
import FillInTheBlanks from '../../FillInTheBlanks/FillInTheBlanks';
import { useAmplitude } from '../../instrumentation/AmplitudeHookProvider';
import { viewPromptDetail } from '../../instrumentation/events';
import { pageTypes } from '../../constants';
import Heading from './Heading';
import tagBySlug from './tagBySlug';

function PromptsPage({ fitbs, history, match: { url }, tags }) {
  const { id, slug } = useParams();
  const tag = tagBySlug({ slug, tags });
  const fitb = fitbs.find(fitb => fitb.id === id);

  const { logEvent } = useAmplitude();
  const { properties, type } = viewPromptDetail;

  useEffect(() => {
    if (fitb) {
      logEvent(type, properties({ fitb, tag, url }));
    }
  }, [fitb, logEvent, properties, tag, type, url]);

  return (
    <>
      <Heading title="TeamPrompts" />
      <Filters filter={{ name: tag.name }} fitbs={fitbs} tags={tags} />
      <ul>
        {fitb &&
          [fitb].map(fitb => (
            <li key={fitb.id}>
              <FillInTheBlanks
                fitb={fitb}
                history={history}
                pageType={pageTypes.PromptsPage}
                tag={tag}
              />
            </li>
          ))}
      </ul>
    </>
  );
}

export default PromptsPage;
