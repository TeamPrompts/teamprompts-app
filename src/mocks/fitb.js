import { SEPARATOR } from '../constants';

export const SOURCE = `Do you know the way to ${SEPARATOR}?`;

const fitb = {
  examples: ['San Jose'],
  id: 'receM4Gus7HH43oLM',
  prompts: ['[place]'],
  source: SOURCE,
  tags: ['recTqD26ElVvXKVad']
};

export default fitb;
