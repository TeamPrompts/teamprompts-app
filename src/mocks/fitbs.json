[
  {
    "author": "John Cutler",
    "examples": [
      "limiting work in progress",
      "it meant that at times people would have nothing to do",
      "limiting WIP made it more possible to work on what was important"
    ],
    "id": "recevUpbRFDbVHx7L",
    "prompts": ["aspect of how you work", "reason", "realization"],
    "source": "I used to find __BLANK__ to be highly counterintuitive. It seemed destined to fail because __BLANK__. I now see things differently. The big shift was realizing that __BLANK__.",
    "tags": ["recTqD26ElVvXKVad"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "increasing",
      "speed of onboarding",
      "increases",
      "rate of overall adoption",
      "their ability to see value earlier"
    ],
    "id": "recglWG0JkW6S4z5T",
    "prompts": [
      "increasing/decreasing",
      "some activity",
      "increases/decreases",
      "key performance metric, financial metric, etc.",
      "factors, caveats, assumptions"
    ],
    "source": "We believe that __BLANK__ __BLANK__ causes __BLANK__ in some __BLANK__. There are a number of assumptions/factors underpinning that causal relationship including __BLANK__.",
    "tags": ["recmdwXtLaj7hls73", "rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "id": "recD5UagMbU0oW464",
    "prompts": [
      "some opportunity",
      "some intervention",
      "know/be more confident",
      "some leading indicator",
      "some approach to de-risking"
    ],
    "source": "If you look at our near-time three month roadmap, you will see a number of bets. The riskiest bet, with the highest potential upside, is our bet to focus on __BLANK__ by trying to __BLANK__. We will __BLANK__ this was the right bet if we observe __BLANK__. We can actively de-risk this bet by __BLANK__.",
    "tags": ["recmdwXtLaj7hls73", "rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "id": "recCWoOnmN4aEi5dT",
    "prompts": [
      "some sea-change in technology adoption and/or some shift in consumer behavior",
      "fast/slower",
      "forcing/allowing",
      "some activity",
      "forcing/allowing",
      "some other activity"
    ],
    "source": "The writing is on the wall about __BLANK__. What is unclear is timing. As a company, we are betting that this will occur __BLANK__ than expected, which will influence our strategy by __BLANK__ us to __BLANK__, and __BLANK__ us to __BLANK__.",
    "tags": ["recmdwXtLaj7hls73", "rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "to add steps to the workflow",
      "more customers completing the existing workflow"
    ],
    "id": "recQ2jud83D1dCxr4",
    "source": "To feel more confident about our decision to __BLANK__, I would like to observe __BLANK__.",
    "tags": ["recG0ANQWrp3SRvvY"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "spend too long deliberating over the behavior of the page filter",
      " test early with customers"
    ],
    "id": "recrYoglrdF32klOv",
    "source": "I’m concerned that if we __BLANK__, we may miss the opportunity to __BLANK__.",
    "tags": ["rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "the commodification of storage",
      "the rate of dropping costs",
      "1) COGS would not drop as fast as we expect"
    ],
    "id": "recm6xCbOJSJZ6jwY",
    "prompts": ["an assumption", "that assumption", "a list of ways"],
    "source": "We seem to be treating __BLANK__ as a foregone conclusion. I’m not sure it is that simple. If we were proven wrong (or less right) about __BLANK__ , it would impact us in the following ways: __BLANK__  .",
    "tags": ["recmdwXtLaj7hls73"]
  },
  {
    "author": "John Doe",
    "examples": ["this amazing challenge"],
    "id": "recqeN7TDq4kdhFah",
    "source": "To even know where to start, we would need to better understand __BLANK__.",
    "tags": ["recG0ANQWrp3SRvvY"]
  },
  {
    "author": "The Person",
    "examples": [
      "this huge problem we have",
      "feel the wrath of all that has come before"
    ],
    "id": "rec99C9lxMl6aSedq",
    "source": "A danger of not resolving __BLANK__ now is that we will eventually __BLANK__.",
    "tags": ["recTqD26ElVvXKVad"]
  },
  {
    "author": "Mary Tine",
    "examples": [
      "how we perceive user onboarding",
      "our projections might be off"
    ],
    "id": "reckkQSV8NuHKOAHr",
    "prompts": ["some assumption", "reason"],
    "source": "The weakest link in our mental model is __BLANK__. If we’re off -- even by a little -- it could have a big impact on this initiative because __BLANK__.",
    "tags": ["rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "understand the EMEA customer persona",
      "we haven't really made EMA a focus",
      "we decide to shift into EMEA"
    ],
    "id": "receM4Gus7HH43oLM",
    "prompts": ["an action", "reason", "new event"],
    "source": "We may be over-confident in our ability to __BLANK__. It hasn’t really been a problem yet because __BLANK__, but it could become a problem in the future if __BLANK__.",
    "tags": ["recmdwXtLaj7hls73"]
  }
]
