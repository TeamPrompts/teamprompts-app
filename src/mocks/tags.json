[
  {
    "fitbs": ["recNsahDzKCx0IQPk"],
    "id": "recmdxpSo7MSWEg00",
    "name": "1:1s",
    "slug": "1-1s"
  },
  {
    "fitbs": ["recQ2jud83D1dCxr4", "recqeN7TDq4kdhFah"],
    "id": "recG0ANQWrp3SRvvY",
    "name": "decision making",
    "slug": "decision-making"
  },
  { "id": "recEZjB6WTnRt6aon", "name": "interview", "slug": "interview" },
  {
    "fitbs": [
      "reckkQSV8NuHKOAHr",
      "recBKiB7u7DLP6QPe",
      "recrYoglrdF32klOv",
      "recCWoOnmN4aEi5dT",
      "recD5UagMbU0oW464",
      "recglWG0JkW6S4z5T"
    ],
    "id": "rechbGc1FLmwxb9Hg",
    "name": "measurement",
    "slug": "measurement"
  },
  {
    "fitbs": ["recnGYhq8E9LeRPlW", "recNsahDzKCx0IQPk"],
    "id": "recBxMIOsQqBhGSaQ",
    "name": "product management",
    "slug": "product-management"
  },
  {
    "fitbs": [
      "rec99C9lxMl6aSedq",
      "rec590aP1VRIE8f23",
      "recNsahDzKCx0IQPk",
      "recevUpbRFDbVHx7L"
    ],
    "id": "recTqD26ElVvXKVad",
    "name": "retrospectives",
    "slug": "retrospectives"
  },
  {
    "id": "rec1f3sLCcKszXSi4",
    "name": "reviewing decisions",
    "slug": "reviewing-decisions"
  },
  {
    "fitbs": [
      "receM4Gus7HH43oLM",
      "recBKiB7u7DLP6QPe",
      "recm6xCbOJSJZ6jwY",
      "recCWoOnmN4aEi5dT",
      "recD5UagMbU0oW464",
      "recglWG0JkW6S4z5T"
    ],
    "id": "recmdwXtLaj7hls73",
    "name": "strategy",
    "slug": "strategy"
  }
]
