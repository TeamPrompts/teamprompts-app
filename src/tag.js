const tag = {
  fitbs: [
    'receM4Gus7HH43oLM',
    'rec99C9lxMl6aSedq',
    'rec590aP1VRIE8f23',
    'recNsahDzKCx0IQPk'
  ],
  id: 'recTqD26ElVvXKVad',
  name: 'retrospectives'
};

export default tag; // INFO: this is a mock
