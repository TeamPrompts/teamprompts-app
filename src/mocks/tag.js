import tags from './tags';

const tag = tags.find((tag) => tag.id === 'recTqD26ElVvXKVad');

export default tag;
