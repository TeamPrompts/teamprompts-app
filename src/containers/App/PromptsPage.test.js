import React from 'react';
import { create } from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import fitb from '../../mocks/fitb';
import fitbs from '../../mocks/fitbs';
import tag from '../../mocks/tag';
import tags from '../../mocks/tags';
import PromptsPage from './PromptsPage';

const mockId = fitb.id;
const mockSlug = tag.slug;

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: mockId,
    slug: mockSlug
  })
}));

describe('PromptsPage', () => {
  it('to match snapshot', () => {
    const tree = create(
      <Router>
        <PromptsPage fitbs={fitbs} tags={tags} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
