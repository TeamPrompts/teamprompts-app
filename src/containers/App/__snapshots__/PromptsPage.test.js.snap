// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromptsPage to match snapshot 1`] = `
Array [
  <div
    className="mt-2"
  >
    <a
      className="text-indigo-700 underline"
      href="#/"
      onClick={[Function]}
    >
      Home
    </a>
     &gt; 
    <a
      className="capitalize text-indigo-700 underline"
      href="#/retrospectives"
      onClick={[Function]}
    >
      retrospectives
    </a>
     &gt; 
    ____
  </div>,
  <h1
    className="font-bold mb-8 mt-2 text-4xl text-gray-900"
  >
    TeamPrompts
  </h1>,
  <ul
    className="flex flex-wrap mb-4"
  >
    <li
      className="mx-1 my-1"
    >
      <a
        href="#/retrospectives"
        onClick={[Function]}
      >
        <button
          className="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-900"
          data-testid="retrospectives"
          onClick={[Function]}
          type="button"
        >
          retrospectives (1)
        </button>
      </a>
    </li>
  </ul>,
  <ul>
    <li>
      <div
        className="border-b border-gray-500 leading-loose py-8 text-lg"
      >
        Do you know the way to 
        <span
          className="inline-flex flex-col leading-none"
        >
          <div
            className="self-center"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              className=""
              data-testid="input"
              onChange={[Function]}
              placeholder="________________"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "1px",
                }
              }
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              ________________
            </div>
          </div>
          <label
            className="self-center text-sm"
          >
            place
          </label>
        </span>
        ?
        <span
          className="inline-block ml-3 relative"
        >
          <input
            checked={false}
            className="hidden"
            data-testid="receM4Gus7HH43oLM"
            id="receM4Gus7HH43oLM"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="cursor-pointer"
            htmlFor="receM4Gus7HH43oLM"
          >
            <div
              className="w-6 h-3 bg-gray-400 rounded-full shadow-inner"
            />
            <div
              className="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0"
            />
          </label>
        </span>
        <button
          className="bg-blue-500 focus:outline-none font-bold font-mono hover:bg-blue-700 ml-3 px-4 rounded-full text-sm text-white"
          data-testid="copy-button"
          onClick={[Function]}
          type="button"
        >
          Ctrl+C
        </button>
        <a
          className="ml-3 text-indigo-700 underline"
          href="#/retrospectives/receM4Gus7HH43oLM"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-link fa-w-16 "
            data-icon="link"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </li>
  </ul>,
]
`;
